src/SonarScanner.MSBuild.Shim/PathHelper.cs

```diff
@@ -52,18 +52,27 @@ public static bool IsInDirectory(this FileInfo file, DirectoryInfo directory)
         }
 
         /// <summary>
-        /// FIXME
+        /// Returns longest common root path.
+        /// In case paths do not share common root, path from most common drive is selected.
         /// </summary>
         public static DirectoryInfo BestCommonRoot(IEnumerable<DirectoryInfo> paths)
         {
-            if (paths == null)
+            if (paths == null || !paths.Any())
+            {
+                return null;
+            }
+            var pathParts = paths.Select(GetParts).ToArray();
+            if (BestRoot(pathParts) is { } bestRoot)
+            {
+                pathParts = pathParts.Where(x => x[0] == bestRoot).ToArray();
```

Comment:
I suggest avoiding to reuse `pathParts` - at this point it might be a subset, so it's better IMO to use a different variable with a suggestive name

Reply:
I still have an itch. `pathParts` is instantiated outside but never used basically except in the `BestRoot` call
how about
```
if (BestRoot(paths.Select(GetParts).ToArray()) is { } bestRoot)
{
  var pathParts = pathParts.Where(x => x[0] == bestRoot).ToArray();
  ...
}
else
{
  return null;
}
```

Reply:
I still have an itch because I favor clarity.

Yes, please use `candidateParts`.

Reply:
how about
```
            var pathParts = paths.Select(GetParts).ToArray();
            if (BestRoot(pathParts) is { } bestRoot
                && pathParts.Where(x => x[0] == bestRoot).ToArray() is var bestRootPaths)
            {
                var shortest = bestRootPaths.OrderBy(x => x.Length).First();
                return new DirectoryInfo(Path.Combine(shortest.TakeWhile((x, index) => bestRootPaths.All(parts => parts[index] == x)).ToArray()));
            }
```

Reply:
I've renamed it to `allPathParts` and `bestRootPathParts` with two variables.

I used standalone `var` statement, as the `is var` pattern would be miss use of the pattern matching feature